karma-cov.conf.js

```diff
@@ -10,16 +10,27 @@ function browser(b) {
   return b.toLowerCase().split(/[ /-]/)[0];
 }
 
+function subdir_name(b) {
+  var subdir = browser(b);
+  if (process.env.GEOJS_TEST_CASE) {
+    var parts = /^(.+\/)*(([^\/]+)\.[^\/.]*|[^\/.]+)$/.exec(process.env.GEOJS_TEST_CASE);
+    if (parts) {
+      subdir += '_' + (parts[3] || parts[2]);
+    }
+  }
+  return subdir;
+}
+
 module.exports = function (config) {
   var karma_config = require('./karma-base')(config);
 
   karma_config.reporters = ['progress', 'coverage'];
   karma_config.coverageReporter = {
     reporters: [
-      {type: 'html', dir: 'dist/coverage/', subdir: browser},
-      {type: 'cobertura', dir: 'dist/cobertura/', file: 'coverage.xml', subdir: browser},
-      {type: 'json', dir: 'dist/coverage/json/', subdir: browser},
-      {type: 'lcovonly', dir: 'lcov', subdir: browser},
+      {type: 'html', dir: 'dist/coverage/', subdir: subdir_name},
+      {type: 'cobertura', dir: 'dist/cobertura/', file: 'coverage.xml', subdir: subdir_name},
```

Comment:
The purpose is to put the coverage for the test groups into different subdirectories.  That said, the regex could use a comment describing what it does.